Pod/Classes/ResultType.swift

```diff
@@ -0,0 +1,123 @@
+//
+//  ResultType.swift
+//  Pods
+//
+//  Created by Bueno McCartney on 12/7/16.
+//
+//
+
+import Foundation
+
+public protocol ResultType {
+    associatedtype Value
+    
+    init(_ value: Value?)
+    init(_ error: Error)
```

Comment:
I actually think this makes it _safer_. This prevents `Result(value: error)` from occurring. I can't imagine a case where that should be valid.